modules/imap/handler_modules.php

```diff
@@ -1304,6 +1314,17 @@ public function process() {
             $this->out('folder_status', $status);
             $this->out('imap_unread_data', $msg_list);
             $this->out('imap_server_ids', $form['imap_server_ids']);
+
+            $webhook_token = $this->user_config->get('webhook_token_setting');
+            if($webhook_token && !empty($webhook_token)) {
+                $extracted_msgs = array();
+                foreach ($msg_list as $msg) {
```

Comment:
I'm not sure this is the right way to do this, a loop: if I receive 50 messages should I therefore receive 50 notifications in Telegram ?

Reply:
@josaphatim, @marclaporte what are your thoughts?

Reply:
I am sure @nekohayo will have some more thoughtful ideas but here is a first one:

Maybe with a 5 minute buffer? Thus: accumulate all the emails, and when no new email has arrived for 5 minutes (configurable), send one message: "You have received 50 emails". If messages don't stop, send one message every 5 minutes.

Reply:
Does the API lets you reuse & edit (or auto-delete/auto-timeout the previous) notification message sent to the chat while the user has not yet read it? Otherwise, other than having some "exponential" back-off (ex: 5 mins, 10, 20, 40, 60, 90...) while messages are not read by the user on chat (if the API allows you to get read receipts, but I kinda doubt it), I don't see a much more creative way than what Marc has proposed here so far.